src/tamp_improv/benchmarks/base.py

```diff
@@ -28,7 +28,9 @@ class PlanningComponents(Generic[ObsType]):
 
     types: set[Type]
     predicate_container: PredicateContainer
-    operators: set[LiftedOperator]
+    base_operators: set[LiftedOperator]  # Without extra preconditions
+    full_operators: set[LiftedOperator]  # With extra preconditions
+    operators: set[LiftedOperator]  # Currently actively set
```

Comment:
it would be slightly better if this dataclass itself makes it impossible for these operators to be something other than the base or full operators, to prevent possible bugs. suggestion: change this to `full_operators_active: bool` and then if you want, you can add a property to the class like

```
@property
def operators(self) -> set[LiftedOperator]:
    return self.full_operators if self.full_operators_active else self.base_operators
```